### ISSUE TITLE
Thank you so much for the update!  In record time too!!!

### ISSUE THREAD

**Issue by gaveitatry** [schomery/dictionary#53]
1.) Best update --> Having the option to translate on the same page without opening new tabs.

2.) Great fix.  Even though I will probably never use browser's extension button to translate, it's good to know it will translate using whatever engine I want.  It was weird right-clicking a page and using Bing, but being forced to use Google translate when using the extension button.

Note: You tried to add a fix to make the iframe warning popup go away, but it doesn't work.  I don't think that this is a failing on your part.  I think that it is 100% impossible to recode your web extension to make those iframe warning popups go away.  It's an AMO policy restriction.  The fact that you might not be getting the popups is probably because you changed something on your browser that cancelled it out.  But the web extension itself does not make it go away.  Best way to test this is to run a vanilla copy of your browser without other addons, user scripts, or about:config changes.  Then try running your newest version extension on a page like https://github.com/Endor8/userChrome.js/tree/master/userChrome and see if you still get the warning popup.  If you can confirm that the extension won't get rid of the popup, you can add that user script that I shared with you earlier in the extension description page so those who download the extension will know a way to get those popups to disappear.

Note:  I think that your extension is like 99% perfect now.  If I were to change anything, it would be two things.  First, I would reword and reformat the wording on the web extension's settings/options page to make it easier to understand (I could try to help with that).  And second, I think that it would be cool if there was an undo option.  If I translate a page, and hit Alt + Left Arrow (or Back button) once, it will just re-translate the page.  The only way to get back to the original untranslated page is to do it twice - Alt + Left Arrow (or Back button) twice.  I noticed that when you first translate the page, you see a translation bar on the top of the screen for a brief moment and then it disappears.  I wanted to play around with that to see if there was a to "undo" translation changes and stuff, but I couldn't figure out how to make that translation bar remain visible and it's probably best to leave it alone since it would take up a lot of space on the translated page and it probably won't do anything useful.  I don't really see a convenient way to restore a page's original content (especially without having to mouse over to the navigation bar extension button, which I don't want to have to do anyway) so I guess just hitting the back button twice might be the best way.  I guess really what I would want is to go back to the original page by only having to hit the back button once, and that would be perfect.  It's intuitive and instinctive of me to hit Alt + Left Arrow once anyway.  The only problem is being forced to do it twice.  If I only had to do it once, then it would feel like I want it to.

Anyway, 99% perfect extension.  Best one for Firefox 57+ (for now).  Thank you for your hard work and dedication to perfection!

**Comment by schomery** [comment]
> Note: You tried to add a fix to make the iframe warning popup go away, but it doesn't work.

Not actually. I noticed that this is out of the scope of this extension and users can fix it if they needed.

> I would reword and reformat the wording on the web extension's settings/options page to make it easier to understand (I could try to help with that)

Sure, please submit as a new bug or prepare a pull merge.

> The only way to get back to the original untranslated page is to do it twice - Alt + Left Arrow (or Back button) twice.

This is out of the scope of this extension to alter the destination page. Though you can simply extend your script to perform `window.history.go(-2);`

**Comment by gaveitatry** [comment]
> "This is out of the scope of this extension to alter the destination page."

Actually, it is not out of the scope of the extension to revert back to the original page by only hitting the back button once.  I know this, because I tried another extension similar to yours called 'Page Translate Revised' by Mathnerd314 (https://addons.mozilla.org/en-US/firefox/addon/page-translator-revised/).  In his extension, if you translate a page (on the same tab), and then hit the back button ONCE it goes back to the original page.  I don't have to hit the back button twice.

> "Though you can simply extend your script to perform window.history.go(-2);"

Do you mean the user script that I created to remove iframe alerts?  I guess that I could give it a try, but it will take a lot of trial and error and I might make a lot of mistakes.  I might not even get it to work.  Or I might end up affecting regular, non translated pages too.  It's too bad that you won't program your extension to do what 'Page Translator Revised' does.  I will give it a try though.  Thanks.

&nbsp;
Of the two extensions, after changes that you both made, both of your extensions are almost of similar quality.  His icon that he uses for his context menu and navigation bar isn't the greatest.  It should have more color.  He also has his context menu as "Translate Page" instead of "Translate this Page".  Those are the two biggest faults of his extension.  And on yours (his doesn't do this), I have to hit the back button twice just to get to the original page.  If he would change "Translate Page" to "Translate this Page", there is a good chance that I would switch to his extension, and then later in the future I will try to find an icon similar to his but that looks better and has more color.  But I doubt that he will ever change the icon because he is probably trying to blend the icon with the URL bar, where it doesn't stand out and detract from other web extension icons.  I don't know.  And if you ever made it so that so that hitting the back button once brought you back to the original page, I would probably stick with your extension for the rest of my life because it would have a prettier icon and do everything that I wanted.  Right now there are minor flaws with your extension and his.  You've improved your extensions some over the last week, but I just wish that you both would "go all the way" and make them even better.  Anyway, I will try to work on the window.history.go(-2) stuff this week.  I will fail and fail again.  But maybe I will get lucky and get it to work.
